traptor/traptor.py

```diff
@@ -863,6 +909,33 @@ def _send_heartbeat_message(self):
             self._add_heartbeat_message_to_redis(self.heartbeat_conn)
             time.sleep(hb_interval)
 
+    @retry(wait=wait_exponential(multiplier=1, max=10),
+           stop=stop_after_attempt(3),
+           reraise=True,
+           retry=retry_if_exception_type(KafkaUnavailableError)
+           )
+    def _send_enriched_data_to_kafka(self, tweet, enriched_data):
+        """"
+        Send the enriched data to Kafka
+
+        :param tweet: the original tweet
+        :param enriched_data: the enriched data to send
+        """
+        try:
+            self.logger.info("Attempting to send tweet to kafka", extra={
+                'tweet_id': tweet.get('id_str', None)
+            })
+            future = self.kafka_conn.send(self.kafka_topic, enriched_data)
+            future.add_callback(self.kafka_success_callback, tweet)
+            future.add_errback(self.kafka_failure_callback)
+        except KafkaUnavailableError:
```

Comment:
Changed back to plain Exception to get the logging. The retry will handle the Kafka error.